src/prefect/projects/steps/core.py

```diff
@@ -75,8 +75,4 @@ async def run_step(step: dict) -> dict:
     inputs = await resolve_variables(inputs)
 
     step_func = _get_function_for_step(fqn, requires=keywords.get("requires"))
-    output = step_func(**inputs)
-    if inspect.isawaitable(output):
-        return await output
-    else:
-        return output
+    return await from_async.wait_for_call_in_new_thread(Call.new(step_func, **inputs))
```

Comment:
You'll only want to use `wait_for_call` if a step function can send work _back_ to this thread. Is that expected? Otherwise you just do `await call_soon...(...).aresult()` and there's less going on behind the scenes.

I'll add a `call_in_new_thread` utility that does this for you in the future and some more docs because it's kind of confusing otherwise.